### PR TITLE
Filter out modular and regular issues when looking for duplicates

### PR DESCRIPTION
**Pull request by antbob**
This patch addresses the issue where we might encounter modular Jira issues that would match as duplicates, for example "postgresql" and "postgresql:15" as component would be the only difference. The Jira cannot distinguish between the two because they map to the exact same component id in Jira thus JQL returning 'duplicate' issues which are in fact not duplicates. The fix borrows the same approach as used for filtering out modular issues already.

### CODE REVIEW

File: ymir/tools/privileged/jira.py [hunk 1]
```diff
@@ -527,8 +537,14 @@ async def _check_duplicate_tracker(
...
     tool = SearchJiraIssuesTool()
-    output = await tool.run(input={"jql": jql, "fields": ["status", "resolution"], "max_results": 50})
-    issues = output.result
+    output = await tool.run(
+        input={"jql": jql, "fields": ["status", "resolution", "summary"], "max_results": 50}
+    )
+    issues = [
+        i
+        for i in output.result
+        if _is_modular_summary(i.get("fields", {}).get("summary", ""), component) == is_modular
```
[the remark this answers is not in the document]
Comment (antbob) [reply ymir/tools/privileged/jira.py]: this cannot happen because Summary is required field but lets handle it anyway just in case

File: ymir/tools/privileged/jira.py [hunk 2]
```diff
@@ -392,12 +392,18 @@ async def _run(
 
 CVE_ID_PATTERN = re.compile(r"(CVE-\d{4}-\d{4,})")
 
+_MODULAR_SUMMARY_PREFIX = r"^(?:CVE-\d{4}-\d+\s+)?([\w.+-]+):([^/\s]+)/"
+
```
[the remark this answers is not in the document]
Comment (antbob) [reply ymir/tools/privileged/jira.py]: moved it to version utils which seem to be the most suitable place for it (no dependencies)

**Review by majamassarini — COMMENTED** [review]
I believe these are valid concerns from claude, at least the first one, the second is an edge case not sure if it will ever happen:

  1. Test silently skipped - testis_modular at ymir/agents/tests/unit/test_triage_agent.py:212 should be test_is_modular (missing underscore). pytest won't discover it without the test_* pattern.
  2. Wrong duplicate detection - jira.py:539: Issues without a summary field get empty string '', which is_modular('', component) returns False for. When filtering for non-modular issues (modular=False), a modular issue missing its summary field would incorrectly match as a duplicate.

**Review by jpodivin — COMMENTED** [review]
Everything seems ok. Except for the issue highlighted by Maja.

**Comment by antbob** [comment]
thanks guys! and sorry about the sloppy test typo. i've fixed both but again the Summary is required field so yeah in practice its unlikely to happen but good to guard against anyway.

**Review by majamassarini — APPROVED** [review]
